Build failure: outils

## Steps To Reproduce

Steps to reproduce the behavior:

1. build *outils*

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
error: builder for '/nix/store/v1j6dj3ghqbwikx0882zki1avjrhnr7b-outils-0.13.drv' failed with exit code 2;
       last 25 log lines:
       > gcc -Wall -O2 -include src/liboutils/outils.h -isystem src/liboutils/include -Isrc/bin/md5 -D_GNU_SOURCE -D_DEFAULT_SOURCE -DNO_UTIL -D_PATH_INCLUDE="\"-I/nix/store/64ww8y70r558zgwhp6hagv0534ka59aw-outils-0.13/share/calendar\""  -c -o src/liboutils/arc4random_uniform.o src/liboutils/arc4random_uniform.c
       > gcc -Wall -O2 -include src/liboutils/outils.h -isystem src/liboutils/include -Isrc/bin/md5 -D_GNU_SOURCE -D_DEFAULT_SOURCE -DNO_UTIL -D_PATH_INCLUDE="\"-I/nix/store/64ww8y70r558zgwhp6hagv0534ka59aw-outils-0.13/share/calendar\""  -c -o src/liboutils/getentropy_linux.o src/liboutils/getentropy_linux.c
       > src/liboutils/getentropy_linux.c:23:9: warning: "_POSIX_C_SOURCE" redefined
       >    23 | #define _POSIX_C_SOURCE 199309L
       >       |         ^~~~~~~~~~~~~~~
       > In file included from /nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/bits/libc-header-start.h:33,
       >                  from /nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/stdio.h:28,
       >                  from ./src/liboutils/outils.h:48,
       >                  from <command-line>:
       > /nix/store/kj8hbqx4ds9qm9mq7hyikxyfwwg13kzj-glibc-2.40-36-dev/include/features.h:300:10: note: this is the location of the previous definition
       >   300 | # define _POSIX_C_SOURCE        200809L
       >       |          ^~~~~~~~~~~~~~~
       > gcc -Wall -O2 -include src/liboutils/outils.h -isystem src/liboutils/include -Isrc/bin/md5 -D_GNU_SOURCE -D_DEFAULT_SOURCE -DNO_UTIL -D_PATH_INCLUDE="\"-I/nix/store/64ww8y70r558zgwhp6hagv0534ka59aw-outils-0.13/share/calendar\""  -c -o src/liboutils/explicit_bzero.o src/liboutils/explicit_bzero.c
       > gcc -Wall -O2 -include src/liboutils/outils.h -isystem src/liboutils/include -Isrc/bin/md5 -D_GNU_SOURCE -D_DEFAULT_SOURCE -DNO_UTIL -D_PATH_INCLUDE="\"-I/nix/store/64ww8y70r558zgwhp6hagv0534ka59aw-outils-0.13/share/calendar\""  -c -o src/liboutils/sha2.o src/liboutils/sha2.c
       > gcc -o src/usr.bin/jot/jot src/usr.bin/jot/jot.o src/liboutils/strlcpy.o src/liboutils/strlcat.o src/liboutils/strtonum.o src/liboutils/arc4random.o src/liboutils/arc4random_uniform.o src/liboutils/getentropy_linux.o src/liboutils/explicit_bzero.o src/liboutils/pledge.o src/liboutils/sha2.o -Wl,--as-needed -lm
       > gcc -Wall -O2 -include src/liboutils/outils.h -isystem src/liboutils/include -Isrc/bin/md5 -D_GNU_SOURCE -D_DEFAULT_SOURCE -DNO_UTIL -D_PATH_INCLUDE="\"-I/nix/store/64ww8y70r558zgwhp6hagv0534ka59aw-outils-0.13/share/calendar\""  -c -o src/usr.bin/lam/lam.o src/usr.bin/lam/lam.c
       > src/usr.bin/lam/lam.c: In function 'getargs':
       > src/usr.bin/lam/lam.c:195:31: error: implicit declaration of function 'recallocarray'; did you mean 'reallocarray'? [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Wimplicit-function-declaration8;;]
       >   195 |                         tmp = recallocarray(input, inputsize,
       >       |                               ^~~~~~~~~~~~~
       >       |                               reallocarray
       > src/usr.bin/lam/lam.c:195:29: error: assignment to 'void *' from 'int' makes pointer from integer without a cast [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wint-conversion-Wint-conversion8;;]
       >   195 |                         tmp = recallocarray(input, inputsize,
       >       |                             ^
       > make: *** [<builtin>: src/usr.bin/lam/lam.o] Error 1
       For full logs, run 'nix log /nix/store/v1j6dj3ghqbwikx0882zki1avjrhnr7b-outils-0.13.drv'.
error: 1 dependencies of derivation '/nix/store/lvvydlj30f62hrq9jpgnlkx1kg4nb5vz-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ik3yah976ap79na6qycwi4vjb6wa3j9f-nixos-system-solid-25.05.20241229.88195a9.drv' failed to build
```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.5, NixOS, 25.05 (Warbler), 25.05.20241217.d3c42f1`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/6zgbbqlr7nnfxpzkyj7fsl4fpg89jbw0-source`

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@somasis 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ou/outils/package.nix
+++ b/pkgs/by-name/ou/outils/package.nix
@@ -1,6 +1,7 @@
 {
   stdenv,
   fetchFromGitHub,
+  fetchpatch2,
   lib,
 }:
 
@@ -14,6 +15,14 @@ stdenv.mkDerivation rec {
     rev = "v${version}";
     hash = "sha256-FokJytwQsbGsryBzyglpb1Hg3wti/CPQTOfIGIz9ThA=";
   };
+
+  patches = [
+    (fetchpatch2 {
+      url = "https://github.com/leahneukirchen/outils/commit/50877e1bf7c905044e0b50b227ecff48cfec394b.patch?full_index=1";
+      name = "outils-add-recallocarray-prototype.patch";
+      hash = "sha256-jOnCMPcHKMRR3J0Yh+ZTHAn7P85FO80yXVX0K2vtlVk=";
+    })
+  ];
 
   makeFlags = [ "PREFIX=$(out)" ];
 

